src/test/java/org/apache/commons/lang3/SystemUtilsTest.java

```diff
@@ -373,6 +373,33 @@ public void testOSMatchesNameAndVersion() {
     }
 
     @Test
+    public void testMatchOSMajorMinorVersion() {
+        String osName = "Mac OS X";
+        String osVersion = null;
+        assertFalse(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        osVersion = "10.10";
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        assertFalse(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.1"));
+        osVersion = "10.1";
+        assertFalse(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.1"));
+        osVersion = "10.10.1";
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        osVersion = "10.10.10";
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+    }
+
+
+    @Test
+    public void testNotConfuseMacOSXPumaAndYosemite() {
```

Comment:
Ah, thinking about this some more I now understand why you couldn't use ```String.startsWith(String)```. Because 10.1 and 10.10 have the same prefix. Well that is a tricky one. I'm not sure I like the proposed solution. Maybe we can make ```isOSMatch```smarter?